gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriter.java

```diff
@@ -353,7 +357,8 @@ public static Path getOutputDir(State state) {
    * Sets the {@link FsPermission}, owner, group for the path passed. It will not throw exceptions, if operations
    * cannot be executed, will warn and continue.
    */
-  public static void safeSetPathPermission(FileSystem fs, FileStatus file, OwnerAndPermission ownerAndPermission) {
+  public static void safeSetPathPermission(FileSystem fs, FileStatus file, OwnerAndPermission ownerAndPermission,
```

Comment:
We should either rename this function, or create a new function, also if it's the former also change the java doc. Currently the API documentation states that it will explicitly warn and continue if operations cannot be executed (which makes sense in some scenarios where permissions should not cause the entire operation to fail), hence `safe` in the function name